File: project/module-script/src/main/java/cn/fd/ratziel/script/SimpleScriptEnv.java
package cn.fd.ratziel.script;

import cn.fd.ratziel.script.api.ScriptEnvironment;
import org.jetbrains.annotations.NotNull;

import javax.script.ScriptContext;
import javax.script.SimpleScriptContext;

/**
 * SimpleScriptEnv
 *
 * @author TheFloodDragon
 * @since 2024/7/15 13:41
 */
public class SimpleScriptEnv implements ScriptEnvironment {

    public SimpleScriptEnv() {
        this(new SimpleScriptContext());
    }

    public SimpleScriptEnv(@NotNull ScriptContext context) {
        this.context = context;
    }

    private final ScriptContext context;

    @Override
    public @NotNull ScriptContext getContext() {
        return context;
    }

}

File: project/module-script/src/main/kotlin/cn/fd/ratziel/module/script/impl/SimpleScriptEnv.kt
package cn.fd.ratziel.module.script.impl

import cn.fd.ratziel.module.script.api.ScriptEnvironment
import javax.script.ScriptContext
import javax.script.SimpleScriptContext

/**
 * SimpleScriptEnv
 *
 * @author TheFloodDragon
 * @since 2025/5/4 15:19
 */
class SimpleScriptEnv(
    private val scriptContext: ScriptContext,
) : ScriptEnvironment {

    constructor() : this(SimpleScriptContext())

    override fun getContext() = this.scriptContext

}